includes/functions/llms.functions.page.php

```diff
@@ -62,37 +62,44 @@ function llms_confirm_payment_url( $order_key = null ) {
 
 }
 
-
 /**
  * Retrieve the full URL to a LifterLMS endpoint
  *
  * @since 1.0.0
  * @since 3.26.3 Unknown.
+ * @since [version] Update to ensure the generated URL has (or doesn't have) a trailing slash based on the site's permalink settings.
  *
  * @param string $endpoint  ID of the endpoint, eg "view-courses".
- * @param string $value     Endpoint query arg value (Optional as the presence of the arg is enough in most scenarios).
+ * @param string $value     Endpoint query parameter value.
  * @param string $permalink Base URL to append the endpoint to. Optional, uses the current page when not supplied.
  * @return string
  */
 function llms_get_endpoint_url( $endpoint, $value = '', $permalink = '' ) {
 
-	if ( ! $permalink ) {
-		$permalink = get_permalink();
-	}
+	$permalink = $permalink ? $permalink : get_permalink();
```

Comment:
Short [ternary operator](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.ternary).
```suggestion
	$permalink = $permalink ?: get_permalink();
```

Reply:
WP forbids the use of short ternary so we do as well: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#ternary-operator

The null coalesce *is allowed*, however.

Reply:
Interesting! Thanks for the reference. I had forgotten that one.